content/blog/federating-spire-on-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,452 @@
+
+## 1.1 Clone the repo using:
+
+```shellsession
+git clone https://github.com/cxteamtrials/caas-trials-content.git 
+```
+
+## 1.2 Apply the QuickStart file on each cluster using the following commands: 
+
+```shellsession
+kubectl apply -f services/spire/federation/spire-quickstart-cluster-1.yaml 
+kubectl apply -f services/spire/federation/spire-quickstart-cluster-2.yaml 
+```
+
+This step will install SPIRE into your clusters, along with two additional components: the SPIFFE CSI Driver and the SPIRE Kubernetes Controller manager which facilitates the registration of workloads and establishment of federated relationships. 
+
+Verify the installation by checking if all pods are running and containers within them are up. 
+
+```shellsession
+Cluster1:~ # kubectl get po -n spire 
+NAME                            READY   STATUS    RESTARTS      AGE
+spire-agent-92q5m               3/3     Running   0             37d
+spire-agent-jhgwf               3/3     Running   0             37d
+spire-agent-sm8gt               3/3     Running   0             37d
+spire-server-574474c7dc-gbzl6   2/2     Running   1 (11d ago)   37d
+```
+
+```shellsession
+Cluster2:~ # kubectl get po -n spire 
+
+NAME                            	 READY      STATUS      RESTARTS       AGE 
+
+spire-agent-wttmd               	 3/3        Running     1 (24h ago)    24h 
+spire-server-574474c7dc-2bfcx   	 2/2        Running     0              24h 
+```
+
+<!--EndFragment-->
+
+<!--StartFragment-->
+
+# Step 2. Installing Istio: 
+
+On each of your clusters, install Istio and patch Istio ingress gateway. Istio can detect the existence of a UNIX Domain Socket that implements the Envoy SDS API on a defined socket path, allowing Envoy to communicate and fetch identities directly from it. SPIRE can be configured for Istio workloads through an integration with Envoy’s SDS API. 
+
+## 2.1 Download the latest release: 
+
+You can download the latest release using the official Istio repository or just copy the following command, which would do the same for you.
+
+```shellsession
+curl -L https://istio.io/downloadIstio | sh - 
+```
+
+Change to the Istio directory (cd command), and set the path by command:
+
+```shellsession
+cd istio-1.17.1 
+export PATH=$PWD/bin:$PATH 
+```
+
+## 2.2 I﻿nstall Istio with custom patch: 
+
+Install Istio with custom patches for the Ingress-gateway as well as for Istio-proxy.  
+
+Get the Istio-spire-config patch using this [link](https://github.com/cxteamtrials/caas-trials-content/blob/main/services/istio/release-1.17/spire), and install that patch using the following commands:
+
+```shellsession
+istioctl apply -f services/istio/release-1.17/spire/spire-patch-cluster1.yaml 
```

Comment:
I would add a comment:
#For K8s cluster 1:
istioctl apply -f services/istio/release-1.17/spire/spire-patch-cluster1.yaml 

#For K8s cluster 2:
istioctl apply -f services/istio/release-1.17/spire/spire-patch-cluster2.yaml